dkg/pedersen/mod.go

```diff
@@ -0,0 +1,258 @@
+package pedersen
+
+import (
+	"go.dedis.ch/fabric/dkg"
+	"go.dedis.ch/fabric/mino"
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/share"
+	"go.dedis.ch/kyber/v3/suites"
+	"go.dedis.ch/kyber/v3/util/random"
+	"golang.org/x/net/context"
+	"golang.org/x/xerrors"
+)
+
+//go:generate protoc -I ./ --go_out=plugins=grpc:./ ./messages.proto
+
+// Factory allows one to create a DKG Starter, which can be used to initialize a
+// new DKG protocol.
+//
+// - implements dkg.Factory
+type Factory struct {
+	pubKeys []kyber.Point
+	privKey kyber.Scalar
+	suite   suites.Suite
+	m       mino.Mino
+}
+
+// NewFactory returns a new DKG Pedersen factory
+func NewFactory(pubKeys []kyber.Point, privKey kyber.Scalar, m mino.Mino,
+	suite suites.Suite) *Factory {
+
+	return &Factory{
+		pubKeys: pubKeys,
+		privKey: privKey,
+		suite:   suite,
+		m:       m,
+	}
+}
+
+// New implements dkg.DKG. It return a new Pedersen Starter.
+func (f *Factory) New(pubKeys []kyber.Point, privKey kyber.Scalar,
+	m mino.Mino, suite suites.Suite) (dkg.Starter, error) {
+
+	h := NewHandler(pubKeys, privKey, m.GetAddressFactory(), m.GetAddress(),
+		suite)
+
+	m, err := m.MakeNamespace("dkg")
+	if err != nil {
+		return nil, xerrors.Errorf("failed to create namespace: %v", err)
+	}
+
+	rpc, err := m.MakeRPC("pedersen", h)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to create RPC: %v", err)
+	}
+
+	return &starter{
+		pubKeys: pubKeys,
+		privKey: privKey,
+		m:       m,
+		suite:   suite,
+		rpc:     rpc,
+	}, nil
+}
+
+// starter allows one to initialise a new DKG protocol.
+//
+// - implements dkg.Starter
+type starter struct {
+	pubKeys []kyber.Point
+	privKey kyber.Scalar
+	m       mino.Mino
+	suite   suites.Suite
+	rpc     mino.RPC
+}
+
+// Start implements dkg.Starter. It allows one to initialize a new DKG protocol.
+func (s *starter) Start(players mino.Players, t uint32) (dkg.DKG, error) {
```

Comment:
The public keys should probably be provided at that point. I guess you need to see with Calypso how you're going to manage the keys but they should basically be stored on the blockchain when integrated to Byzcoin.